src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1259 @@
+
+    def defining_polynomials(self):
+        r"""
+        Return the defining polynomials of the elements of ``self``.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 - x^3, x, y, y^2 + x * y + x^2)
+            sage: A.defining_polynomials()
+            (-x^3 + y^2, x, y, x^2 + x*y + y^2)
+        """
+        return tuple(h.defining_polynomial() for h in self)
+
+    def defining_polynomial(self, simplified=True):
+        r"""
+        Return the defining polynomial of the union of the curves in ``self``.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y ** 2 + x ** 2, x, y)
+            sage: prod(A.defining_polynomials()) == A.defining_polynomial()
+            True
+        """
+        return prod(self.defining_polynomials())
+
+    def have_common_factors(self):
+        r"""
+        Check if the curves have common factors.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(x * y, x^2 + x* y^3)
+            sage: A.have_common_factors()
+            True
+            sage: H(x * y, x + y^3).have_common_factors()
+            False
+        """
+        L = [c.defining_polynomial() for c in self]
+        C = Combinations(L, 2)
+        for f1, f2 in C:
+            if f1.gcd(f2).degree() > 0:
+                return True
+        return False
+
+    def reduce(self, clean=False):
+        r"""
+        Replace the curves by their reduction.
+
+        INPUT:
+
+        - ``clean`` -- boolean (default: False); if ``False``
+          and there are common factors it returns ``None`` and
+          a warning message. If ``True``, the common factors are kept
+          only in the first occurance.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y^2, (x + y)^3 * (x^2 + x * y + y^2))
+            sage: A.reduce()
+            Arrangement (y, x^3 + 2*x^2*y + 2*x*y^2 + y^3) in Affine Space
+            of dimension 2 over Rational Field
+            sage: C = H(x*y, x*(y + 1))
+            sage: C.reduce()
+            Some curves have common components
+            sage: C.reduce(clean=True)
+            Arrangement (x*y, y + 1) in Affine Space of dimension 2
+            over Rational Field
+            sage: C = H(x*y, x)
+            sage: C.reduce(clean=True)
+            Arrangement (x*y) in Affine Space of dimension 2 over Rational Field
+        """
+        P = self.parent()
+        R = self.coordinate_ring()
+        L = [self[0].defining_polynomial().radical()]
+        for c in self[1:]:
+            g = c.defining_polynomial().radical()
+            for f in L:
+                d = g.gcd(f)
+                if d.degree() > 0 and not clean:
+                    print("Some curves have common components")
+                    return None
+                g = R(g / d)
```

Comment:
```suggestion
                g //= d
```
this will return an object in the same ring.